src/api/services/adminServiceClient.ts

```diff
@@ -86,15 +98,10 @@ export const getApiKeysBySite = async (siteId: number): Promise<ApiKeyAdmin[]> =
   return response.data;
 };
 
-export const getApiKeyById = async (keyId: String): Promise<ApiKeyAdmin | undefined> => {
+export const getApiKeyById = async (keyId: String): Promise<ApiKeyAdmin> => {
   const response = await adminServiceClient.get<ApiKeyAdmin>(`/api/client/keyId`, {
     params: { keyId },
   });
-
-  if (response.status === 404) {
-    return undefined;
-  }
-
```

Comment:
This code is never hit because it will go to fallback handler during the code above